modules/rgbd/test/test_pose_graph.cpp

```diff
@@ -0,0 +1,159 @@
+static Ptr<kinfu::detail::PoseGraph> readG2OFile(const std::string& g2oFileName)
+{
+    Ptr<kinfu::detail::PoseGraph> pg = kinfu::detail::PoseGraph::create();
+
+    // for debugging purposes
+    size_t minId = 0, maxId = 1 << 30;
+
+    std::ifstream infile(g2oFileName.c_str());
+    if (!infile)
+    {
+        CV_Error(cv::Error::StsError, "failed to open file");
+    }
+
+    while (infile.good())
+    {
+        std::string data_type;
+        // Read whether the type is a node or a constraint
+        infile >> data_type;
+        if (data_type == "VERTEX_SE3:QUAT")
+        {
+            size_t id;
+            infile >> id;
+            Affine3d pose = readAffine(infile);
+
+            if (id < minId || id >= maxId)
+                continue;
+
+            bool fixed = (id == minId);
+
+            // Ensure we don't have duplicate poses
+            if (pg->isNodeExist(id))
+            {
+                CV_LOG_INFO(NULL, "duplicated node, id=" << id);
+            }
+            pg->addNode(id, pose, fixed);
+        }
+        else if (data_type == "EDGE_SE3:QUAT")
+        {
+            size_t startId, endId;
+            infile >> startId >> endId;
+            Affine3d pose = readAffine(infile);
+
+            Matx66d info;
+            for (int i = 0; i < 6 && infile.good(); ++i)
+            {
+                for (int j = i; j < 6 && infile.good(); ++j)
+                {
+                    infile >> info(i, j);
+                    if (i != j)
+                    {
+                        info(j, i) = info(i, j);
+                    }
+                }
+            }
+
+            if ((startId >= minId && startId < maxId) && (endId >= minId && endId < maxId))
+            {
+                pg->addEdge(startId, endId, pose, info);
+            }
+        }
+        else
+        {
+            CV_Error(cv::Error::StsError, "unknown tag");
+        }
+
+        // Clear any trailing whitespace from the line
+        infile >> std::ws;
+    }
+
+    return pg;
+}
+
+// Turn it on if you want to see resulting pose graph nodes
+static const bool writeToObjFile = false;
+// Turn if off if you don't need log messages
+static const bool verbose = true;
+
+#if !defined(_DEBUG) && defined(HAVE_EIGEN)
+TEST( PoseGraph, sphereG2O )
+{
+    // The dataset was taken from here: https://lucacarlone.mit.edu/datasets/
+    // Connected paper:
+    // L.Carlone, R.Tron, K.Daniilidis, and F.Dellaert.
+    // Initialization Techniques for 3D SLAM : a Survey on Rotation Estimation and its Use in Pose Graph Optimization.
+    // In IEEE Intl.Conf.on Robotics and Automation(ICRA), pages 4597 - 4604, 2015.
+
+    std::string filename = cvtest::TS::ptr()->get_data_path() + "rgbd/sphere_bignoise_vertex3.g2o";
+    Ptr<kinfu::detail::PoseGraph> pg = readG2OFile(filename);
+
+    if (verbose)
+    {
+        cv::utils::logging::setLogLevel(cv::utils::logging::LogLevel::LOG_LEVEL_INFO);
+    }
```

Comment:
Please do not touch logging level in tests.
It may be configured externally, e.g. `OPENCV_LOG_LEVEL=INFO` environment (so even `verbose` parameter is not needed and should be removed)